sipn_reanalysis_ingest/luigitasks/convert.py

```diff
@@ -38,67 +44,112 @@ class Grib2ToNcDaily(luigi.Task):
-                window_start=five_day_window.start,
-                window_end=five_day_window.end,
-                product_type=CfsrGranuleProductType.FORECAST,
-            ),
-        }
-
-        # If the first day of analysis window, we also need the forecast files for the
-        # previous window!
-        if self.date == five_day_window.start:
-            forecast_5day_window = Cfsr5ishDayWindow.from_date_in_window(
-                self.date - dt.timedelta(days=1)
-            )
-            req[CfsrGranuleProductType.FORECAST] = [
-                req[CfsrGranuleProductType.FORECAST],
-                UntarCfsr5DayFile(
-                    window_start=forecast_5day_window.start,
-                    window_end=forecast_5day_window.end,
+        if self.uses_5day_tars:
+            five_day_window = Cfsr5ishDayWindow.from_date_in_window(self.date)
+
+            req = {
+                CfsrGranuleProductType.ANALYSIS: UntarCfsr5DayFile(
+                    window_start=five_day_window.start,
+                    window_end=five_day_window.end,
+                    product_type=CfsrGranuleProductType.ANALYSIS,
+                ),
+                CfsrGranuleProductType.FORECAST: UntarCfsr5DayFile(
+                    window_start=five_day_window.start,
+                    window_end=five_day_window.end,
                     product_type=CfsrGranuleProductType.FORECAST,
                 ),
-            ]
+            }
 
-        return req
+            # If the first day of analysis window, we also need the forecast files for the
+            # previous window!
+            if self.date == five_day_window.start:
+                forecast_5day_window = Cfsr5ishDayWindow.from_date_in_window(
+                    self.date - dt.timedelta(days=1)
+                )
+                req[CfsrGranuleProductType.FORECAST] = [
+                    req[CfsrGranuleProductType.FORECAST],
+                    UntarCfsr5DayFile(
+                        window_start=forecast_5day_window.start,
+                        window_end=forecast_5day_window.end,
+                        product_type=CfsrGranuleProductType.FORECAST,
+                    ),
+                ]
+
+            return req
+        else:
+            # FIXME: Handle the case of the 1st day of daily data. We need to grab a
+            # 5-day and 1-day file....
+            # Don't need to worry about the 5-day window after arbritrary cutoff date
+            req = {
+                CfsrGranuleProductType.ANALYSIS: UntarCfsr1DayFile(
+                    window_start=self.date,
+                    window_end=self.date,
+                    product_type=CfsrGranuleProductType.ANALYSIS,
+                ),
+                CfsrGranuleProductType.FORECAST: UntarCfsr1DayFile(
+                    window_start=self.date - dt.timedelta(days=1),
+                    window_end=self.date,
+                    product_type=CfsrGranuleProductType.FORECAST,
+                )
```

Comment:
Instead of using a window, what about associating each class instance with a single date?

Also, we get both analysis and forecast data in one tar file, so we don't need to specify product type in the dict keys or in the parameters to the task class.